# Titanic Dataset - Exploratory Data Analysis

## Objective
This is my first EDA project in the neurofive-ml-track. I'm exploring the Titanic dataset to understand its structure, identify missing values, and get comfortable with data analysis in Python. This is the foundation before any machine learning work.

In [3]:
# Import the libraries we need
# pandas helps us work with tabular data (like spreadsheets)
# numpy helps with numerical operations
import pandas as pd
import numpy as np

print("Libraries loaded successfully!")

Libraries loaded successfully!


In [5]:
# Load the dataset from the data folder (going up one level)
titanic = pd.read_csv('../data/train.csv')

# Show the first 5 rows
titanic.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


## First Look at the Data
The table above shows the first 5 passengers on the Titanic. I can see:
- Each row is one passenger
- Columns include PassengerId, Survived (0 = didn't survive, 1 = survived), Name, Sex, Age, and Fare
- Survived is the target variable we might want to predict later

In [6]:
# Get information about the dataset
# .info() tells us about column types and missing values
titanic.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


## Dataset Information
Looking at the output above:
- There are 891 rows (passengers)
- Some columns have missing values:
  - Age has only 714 non-null values (177 missing)
  - Cabin has only 204 non-null values (687 missing!)
  - Embarked has 889 non-null values (2 missing)
- Data types: int64, float64, object (text)

In [7]:
# Get statistical summary for numerical columns
# .describe() shows mean, min, max, quartiles, etc.
titanic.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,714.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.699118,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,14.526497,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,20.125000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,38.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


## Statistical Summary
The numbers above tell me:
- Average age was about 29.7 years
- Fares ranged from 0 to 512 (some very expensive tickets!)
- About 38% of passengers survived (mean of Survived is 0.38)
- The youngest passenger was 0.42 years old (a baby!) and oldest was 80

In [8]:
# Let's identify specific information about the dataset

# 1. Total rows and columns
print(f"Total rows: {titanic.shape[0]}")
print(f"Total columns: {titanic.shape[1]}")

# 2. Which columns have missing values?
print("\nMissing values in each column:")
print(titanic.isnull().sum())

# 3. Which columns are numerical?
numerical_cols = titanic.select_dtypes(include=[np.number]).columns.tolist()
print(f"\nNumerical columns: {numerical_cols}")

# 4. Which columns are categorical?
# Using both 'object' and 'string' to work with current and future pandas versions
categorical_cols = titanic.select_dtypes(include=['object', 'string']).columns.tolist()
print(f"\nCategorical columns: {categorical_cols}")

Total rows: 891
Total columns: 12

Missing values in each column:
PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Numerical columns: ['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']

Categorical columns: ['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']


## My Findings (Data Story)

After exploring the Titanic dataset, here's what I discovered:

The dataset has 891 passengers and 12 columns of information. The biggest surprise was the missing data - especially in the Cabin column (687 missing values!) and Age column (177 missing). This is really important to know before building any models.

The data is a mix of numbers and text. The numerical columns include Age, Fare, and family-related info (SibSp, Parch). The categorical columns include Sex, Embarked, and Name.

I noticed that Name could be useful because it has titles like Mr., Mrs., or Miss. The Cabin column has so many missing values that it might be hard to use effectively.

This was my first time doing EDA on a real dataset, and I can already see why it's so important. You need to understand your data before you can analyze it properly. Next steps would be handling the missing values and maybe creating new features from the existing ones.

In [ ]:
# Quick check of all the files in the data folder
import os

print("Files in data folder:")
for file in os.listdir('data/'):
    if file.endswith('.csv'):
        print(f"- {file}")

# Load test.csv just to see its structure (no survival column)
test_data = pd.read_csv('data/test.csv')
print("\nTest data shape:", test_data.shape)
print("Test data columns:", test_data.columns.tolist())

Files in data folder:
- gender_submission.csv
- test.csv
- train.csv

Test data shape: (418, 11)
Test data columns: ['PassengerId', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked']


## All Dataset Files
The data folder contains:
- train.csv - Our main dataset with survival information (891 rows)
- test.csv - Test data without survival column (418 rows)
- gender_submission.csv - Sample submission format for Kaggle

For this EDA, I'm focusing on train.csv since it has the survival data we need to understand.